### PR TITLE
Chore: Update to node 20

### LINKED ISSUE #74533: @grafana/data datetime tests fail in node >18.13.0
### What happened?

Frontend tests are failing on ppc64le:

`yarn test --watchAll==false
jest-haste-map: duplicate manual mock found: store.navIndex.mock
  The following files share their name; please delete one of them:
    * <rootDir>/public/app/features/connections/__mocks__/store.navIndex.mock.ts
    * <rootDir>/public/app/features/datasources/__mocks__/store.navIndex.mock.ts

jest-haste-map: duplicate manual mock found: index
  The following files share their name; please delete one of them:
    * <rootDir>/public/app/features/datasources/__mocks__/index.ts
    * <rootDir>/public/app/features/plugins/admin/__mocks__/index.ts

 FAIL  packages/grafana-data/src/datetime/formats.test.ts
  ● Date Formats › localTimeFormat

    expect(received).toBe(expected) // Object.is equality

    Expected: "MM/DD/YYYY, hh:mm:ss A"
    Received: "MM/DD/YYYY, hh:mm:ss A"

      15 |     );
      16 |
    > 17 |     expect(format).toBe('MM/DD/YYYY, hh:mm:ss A');
         |                    ^
      18 |   });
      19 | });
      20 |

      at Object.<anonymous> (packages/grafana-data/src/datetime/formats.test.ts:17:20)

  ● systemDateFormats › contains correct browser-localized date formats

    expect(received).toBe(expected) // Object.is equality

    Expected: "MM/DD/YYYY, hh:mm:ss A"
    Received: "MM/DD/YYYY, hh:mm:ss A"

      53 |   it('contains correct browser-localized date formats', () => {
      54 |     systemDateFormats.useBrowserLocale();
    > 55 |     expect(systemDateFormats.fullDate).toBe('MM/DD/YYYY, hh:mm:ss A');
         |                                        ^
      56 |     expect(systemDateFormats.fullDateMS).toBe('MM/DD/YYYY, hh:mm:ss.SSS A');
      57 |     expect(systemDateFormats.interval.millisecond).toBe('HH:mm:ss.SSS');
      58 |     expect(systemDateFormats.interval.second).toBe('HH:mm:ss');

      at Object.<anonymous> (packages/grafana-data/src/datetime/formats.test.ts:55:40)

 FAIL  public/app/core/components/SharedPreferences/SharedPreferences.test.tsx (15.914 s)
  ● SharedPreferences › saves the users new preferences

    Timed out in waitFor.

       8 |   input: HTMLElement,
       9 |   optionOrOptions: string | RegExp | Array<string | RegExp>
    > 10 | ) => await waitFor(() => select(input, optionOrOptions, { container: document.body }));
         |                   ^
      11 |
      12 | // Finds the parent of the Select so you can assert if it has a value
      13 | export const getSelectParent = (input: HTMLElement) =>

      at waitForWrapper (.yarn/cache/@testing-library-dom-npm-9.3.1-ec81dc9367-8ee3136451.zip/node_modules/@testing-library/dom/dist/wait-for.js:160:27)
      at public/test/helpers/selectOptionInTest.ts:10:19
      at .yarn/cache/tslib-npm-2.6.0-4d336a6824-c01066038f.zip/node_modules/tslib/tslib.js:169:75
      at Object.__awaiter (.yarn/cache/tslib-npm-2.6.0-4d336a6824-c01066038f.zip/node_modules/tslib/tslib.js:165:16)
      at selectOptionInTest (public/test/helpers/selectOptionInTest.ts:10:5)
      at public/app/core/components/SharedPreferences/SharedPreferences.test.tsx:157:29
      at fulfilled (.yarn/cache/tslib-npm-2.6.0-4d336a6824-c01066038f.zip/node_modules/tslib/tslib.js:166:62)`

### What did you expect to happen?

All tests suites should have passed.

### How do we reproduce it?
run yarn test --watchAll==false

### Environment (with versions)?

Grafana version: v10.1.1
User OS: RHEL 8.7
Arch: ppc64le
Any pointers to resolve this issue? Thanks!

### PR DESCRIPTION
**Pull request by jackw**
**What is this feature?**

This PR updates Grafana to use Node 20.

**Why do we need this feature?**

To keep us up to date and make sure we support Node LTS. Node 20 will go into LTS on 25th Oct.

**Who is this feature for?**

Grafana developers

**Which issue(s) does this PR fix?**:

Fixes https://github.com/grafana/grafana/issues/74533

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

**Review by kminehart — APPROVED** [review]
Technically the build container changes won't do anything but we might as well update it. It's still being used in some small steps not related to building stuff.

File: packages/grafana-data/src/datetime/formats.test.ts [hunk 1]
```diff
@@ -52,8 +52,8 @@ describe('systemDateFormats', () => {
 
   it('contains correct browser-localized date formats', () => {
     systemDateFormats.useBrowserLocale();
-    expect(systemDateFormats.fullDate).toBe('MM/DD/YYYY, hh:mm:ss A');
-    expect(systemDateFormats.fullDateMS).toBe('MM/DD/YYYY, hh:mm:ss.SSS A');
+    expect(systemDateFormats.fullDate).toBe('MM/DD/YYYY, hh:mm:ss A');
+    expect(systemDateFormats.fullDateMS).toBe('MM/DD/YYYY, hh:mm:ss.SSS A');
```
Comment (jackw) [inline packages/grafana-data/src/datetime/formats.test.ts:56]: Please see https://github.com/grafana/grafana/issues/74533#issuecomment-1715830938 for explanation of breaking change in node >=18.13.0 and date formats.

**Review by academo — APPROVED** [review]
Tested like this:

* Checkout to branch
* run `nvm use`. `node --version=v20.8.0`
* run `yarn` . `yarn --version=3.6.4`
* run `yarn build`: All good
* run `yarn typecheck`: All good
* run `yarn lint`: all good
* run `yarn packages:build`: all good
* run `yarn packages:pack`: all good
* run `yarn betterer`: all good
* run `yarn jest`: Got some broken tests non related to nodejs changes. Flaky tests.

I tried to run many nodejs-related commands in the repository without problems and it all seems to work

`yarn packages:pack` generates archives for npm artifacts with the correct `publishConfig`, `files`,`module` and `main` values. 

nice work @jackw !

**Review by leventebalogh — APPROVED** [review]
I didn't test it locally yet either, but the changes look good to me 👍

**Review by ashharrison90 — APPROVED** [review]
looks great! 🙌 

i marked it as fixing https://github.com/grafana/grafana/issues/74533 as well

**Comment by jackw** [comment]
Note: do not merge this until node 20 is LTS. https://nodejs.dev/en/about/releases/

**Comment by chenrui333** [comment]
@jackw looks like this change is not tagged in 10.2.2, is that desired? Thanks!

relates to https://github.com/Homebrew/homebrew-core/pull/155016

**Pull request merged**